src/utils/map-state.js

```diff
@@ -278,10 +293,16 @@ export default class MapState {
   }
 
   // Apply any constraints (mathematical or defined by _viewportProps) to map state
+  /* eslint-disable complexity */
   _applyConstraints(props) {
     // Normalize degrees
-    props.longitude = mod(props.longitude + 180, 360) - 180;
-    props.bearing = mod(props.bearing + 180, 360) - 180;
+    const {longitude, bearing} = props;
+    if (longitude < -180 || longitude > 180) {
+      props.longitude = mod(longitude + 180, 360) - 180;
+    }
+    if (bearing < -180 || bearing > 180) {
+      props.bearing = mod(bearing + 180, 360) - 180;
+    }
```

Comment:
These changes are to avoid changing `longitude` and `bearing` if unnecessary. Otherwise precision issue will break TransitionManager's `_areViewportsEqual`.

Reply:
@1chandu the linear interpolator needs to handle longitude and bearing change cross the 180 boundary.

Reply:
> 3. In a 'reactive" environment we should of course take care not to modify input values unless we have to, but it seems to me it should not matter for viewport equality as that (i.e. _areViewportsEqual) should be done on rounded values? math.gl exports an equals(a,b) that could be used.

I though I had code in interpolator methods that return end viewport when time = 1.0 (In transition manager I enforce interpolation to end with time = 1.0), but looks like I may have moved that code, I will double check and add it back if not present.

And yes I agree,  we should use `equals(a,b)` to avoid any rounding issues.